core/src/main/scala/chisel3/When.scala

```diff
@@ -111,7 +111,9 @@ final class WhenContext private[chisel3] (
   def otherwise(block: => Any)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Unit =
     new WhenContext(sourceInfo, None, block, firrtlDepth + 1, cond ++: altConds)
 
-  def active(): Boolean = scopeOpen
+  def active: Boolean = scopeOpen
+  @deprecated("Calling a nullary function with an empty argument list will be removed by official release", "chisel3")
+  def active(dummy: Int = 0): Boolean = active
```

Comment:
why did you sometimes use this Int-with-default form and sometimes use the variable argument list?